src/mongoose_user_cache.erl

```diff
@@ -0,0 +1,101 @@
+-module(mongoose_user_cache).
+
+-include("mongoose_config_spec.hrl").
+
+-export([start_new_cache/3, stop_cache/2, handle_telemetry_event/4,
+         config_spec/0, process_cache_config/1]).
+
+-export([is_member/3, get_entry/3, merge_entry/4, delete_user/3, delete_domain/3]).
+
+-spec is_member(mongooseim:host_type(), module(), jid:jid()) -> boolean().
+is_member(HostType, Module, Jid) ->
+    CacheName = cache_name(HostType, Module),
+    segmented_cache:is_member(CacheName, key(Jid)).
+
+-spec get_entry(mongooseim:host_type(), module(), jid:jid()) -> term() | not_found.
+get_entry(HostType, Module, Jid) ->
+    CacheName = cache_name(HostType, Module),
+    segmented_cache:get_entry(CacheName, key(Jid)).
+
+-spec merge_entry(mongooseim:host_type(), module(), jid:jid(), map()) -> boolean().
+merge_entry(HostType, Module, Jid, Entry) ->
+    CacheName = cache_name(HostType, Module),
+    segmented_cache:merge_entry(CacheName, key(Jid), Entry).
+
+-spec delete_user(mongooseim:host_type(), module(), jid:jid()) -> true.
+delete_user(HostType, Module, Jid) ->
+    CacheName = cache_name(HostType, Module),
+    segmented_cache:delete_entry(CacheName, key(Jid)).
+
+-spec delete_domain(mongooseim:host_type(), module(), jid:lserver()) -> true.
+delete_domain(HostType, Module, Domain) ->
+    CacheName = cache_name(HostType, Module),
+    segmented_cache:delete_pattern(CacheName, {{'_', Domain}, '_'}).
+
+
+%% path: (host_config[].)modules.*.cache
+config_spec() ->
+    #section{
+       items = #{<<"module">> => #option{type = atom, validate = {enum, [internal, mod_cache_users]}},
```

Comment:
It is not a blocker, but I would omit this option in `mod_cache_users` as it does not make sense there, jsut for better clarity - it confused me a bit when reading the code.